File: backend/src/main/java/lt/techin/lsf/model/Category.java
package lt.techin.lsf.model;

import lombok.*;

import java.util.UUID;

@Getter
@Setter
@Builder
@RequiredArgsConstructor
@AllArgsConstructor
public class Category {
    private UUID uuid;
    private String nameLt;
    private String nameEn;
    private String descriptionLt;
    private String descriptionEn;
    private String albumType;
    private int photoLimit;
    private UUID competitionUuid;
}

File: frontend/src/components/utils/Category.js
import i18n from "../../modules/language/i18n";

class Category {
    constructor(data) {
        this.data = data;
    }

    getUuid() {
        return this.data.uuid;
    }

    getName() {
        switch (i18n.language) {
            case "en":
                return this.data.nameEn;
            case "lt":
            default:
                return this.data.nameLt;
        }
    }

    getDescription() {
        switch (i18n.language) {
            case "en":
                return this.data.descriptionEn;
            case "lt":
            default:
                return this.data.descriptionLt;
        }
    }
}

export default Category;